tensorbay/opendataset/UrbanObjectDetection/loader.py

```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/env python3
+#
+# Copyright 2021 Graviti. Licensed under MIT License.
+#
+# pylint: disable=invalid-name, missing-module-docstring
+
+"""Dataloader of UrbanObjectDetection dataset."""
+
+import os
+
+from tensorbay.dataset import Data, Dataset
+from tensorbay.label import LabeledBox2D
+
+try:
+    import xmltodict
+except ModuleNotFoundError:
+    from tensorbay.opendataset._utility.mocker import xmltodict  # pylint:disable=ungrouped-imports
+
+_SEGMENT_NAMES = ("train", "val", "test", "trainval")
+DATASET_NAME = "UrbanObjectDetection"
+
+
+def UrbanObjectDetection(path: str) -> Dataset:
+    """`UrbanObjectDetection <http://www.rovit.ua.es/dataset/traffic/>`_ dataset.
+
+    The file structure should be like::
+
+        <path>
+            Annotations/
+                <image_name>.xml
+                ...
+            JPEGImages/
+                <image_name>.jpg
+                ...
+            ImageSets/
+                    train.txt
+                    val.txt
+                    test.txt
+                    trainval.txt
+                    ...
+                ...
+            ...
+
+    Arguments:
+        path: The root directory of the dataset.
+
+    Returns:
+        Loaded :class: `~tensorbay.dataset.dataset.Dataset` instance.
+
+    """
+    root_path = os.path.abspath(os.path.expanduser(path))
+    annotation_path = os.path.join(root_path, "Annotations")
+    image_path = os.path.join(root_path, "JPEGImages")
+    main_path = os.path.join(root_path, "ImageSets")
+
+    dataset = Dataset(DATASET_NAME)
+    dataset.load_catalog(os.path.join(os.path.dirname(__file__), "catalog.json"))
+
+    for segment_name in _SEGMENT_NAMES:
+        segment = dataset.create_segment(segment_name)
+        with open(os.path.join(main_path, f"{segment_name}.txt"), encoding="utf-8") as fp:
+            for filename in fp:
+                segment.append(_get_data(filename.rstrip(), image_path, annotation_path))
+    return dataset
+
+
+def _get_data(filename: str, image_path: str, annotation_path: str) -> Data:
```

Comment:
Is it possible to reuse the `_get_data` function in VOC2012Detection loader?
Maybe add a `voc.py` file in `opendataset/_utility` and put the `get_data` function in it?

Reply:
it is ok to reuse, i test it on the whole dataset. do I need to change VOC2012Detection loader?